extensions/tika/runtime/src/main/java/io/quarkus/tika/runtime/TikaRecorder.java

```diff
@@ -0,0 +1,42 @@
+package io.quarkus.tika.runtime;
+
+import java.io.InputStream;
+
+import org.apache.tika.config.TikaConfig;
+
+import io.quarkus.arc.runtime.BeanContainer;
+import io.quarkus.runtime.annotations.Recorder;
+import io.quarkus.tika.TikaParseException;
+import io.quarkus.tika.TikaParser;
+
+@Recorder
+public class TikaRecorder {
+    private static volatile TikaParser parser;
+
+    public void initTikaParser(BeanContainer container, TikaConfiguration config) {
+        initialize(config);
+        TikaParserProducer producer = container.instance(TikaParserProducer.class);
+        producer.initialize(parser);
+    }
+
+    private void initialize(TikaConfiguration config) {
+        if (parser == null) {
+            // tika-config.xml resource
+            TikaConfig tikaConfig = null;
+            InputStream is = getClass().getResourceAsStream(
```

Comment:
Are you sure the `InputStream` is properly closed? This seems a bit fragile to me.